Core/include/Acts/EventData/SpacePointContainer.hpp

```diff
@@ -0,0 +1,160 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/EventData/SpacePointData.hpp"
+#include "Acts/EventData/SpacePointProxy.hpp"
+#include "Acts/EventData/SpacePointProxyIterator.hpp"
+#include "Acts/EventData/Utils.hpp"
+#include "Acts/Utilities/HashedString.hpp"
+
+#include <any>
+#include <vector>
+
+#include <math.h>
+
+namespace Acts {
+struct SpacePointContainerConfig {
+  bool useDetailedDoubleMeasurementInfo = false;
+  bool isInInternalUnits = false;
```

Comment:
can you elaborate? I'm doing this way to be in sync with all the other configs and options objects we have around ACTS

Reply:
I wondered why we have the `isInInternalUnits` param because usually we require everything to be in our internal units. Making this configurable breaks this assumption. Can't the unit conversion just happen on the user side when interacting with those values?

Reply:
As discussed privately, this is an object design point that goes outside this PR scope. For now I'm keeping it this way.